addons/knowledge/static/src/js/widgets/knowledge_dialogs.js

```diff
@@ -0,0 +1,186 @@
+odoo.define('knowledge.dialogs', function (require) {
+'use strict';
+
+const Dialog = require('web.Dialog');
+const { _t } = require('web.core');
+
+const MoveArticleToDialog = Dialog.extend({
+    template: 'knowledge.knowledge_move_article_to_modal',
+    /**
+     * @override
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {Object} props
+     */
+    init: function (parent, options, props) {
+        // Set default options:
+        options.title = options.title || _t('Move an Article');
+        options.buttons = options.buttons || [{
+            text: _t('Ok'),
+            classes: 'btn-primary',
+            click: this.save.bind(this)
+        }, {
+            text: _t('Cancel'),
+            close: true
+        }];
+        this._super(...arguments);
+        this.props = props;
+        // Set template variables:
+        const { state } = props;
+        this.article_name = state.data.name;
+    },
+
+    /**
+     * @override
+     * @returns
+     */
+    start: async function () {
+        const result = await this._super(...arguments);
+        this.initSelect2();
+        return result;
+    },
+
+    /**
+     * @returns {JQuery}
+     */
+    getInput: function () {
+        return this.$el.find('input');
+    },
+
+    /**
+     * @returns {String}
+     */
+    getLoggedUserPicture: function () {
+        const { context } = this.props.state;
+        const origin = window.location.origin;
+        return `${origin}/web/image?model=res.users&field=avatar_128&id=${context.uid}`;
+    },
+
+    initSelect2: function () {
+        const cache = {
+            results: [{
+                text: _t('Categories'),
+                children: [{
+                    id: 'private',
```

Comment:
Not really an id, more like a category_type?

Reply:
For the "id", then use "private_category" and "workspace_category" to have clear naming?

> Note that the select2 library expects the results to be formatted in a certain way.

Could be explained in the docstring to ease reading.